src/irmin.mli

```diff
@@ -1796,75 +1575,625 @@ module Private: sig
+
+  type step
+  (** The type for {!key} steps. *)
+
+  type key
+  (** The type for store keys. A key is a sequence of {!step}s. *)
+
+  type metadata
+  (** The type for store metadata. *)
+
+  type contents
+  (** The type for store contents. *)
+
+  type node
+  (** The type for store nodes. *)
+
+  type tree = [ `Empty | `Node of node | `Contents of contents * metadata ]
+  (** The type for store trees. *)
+
+  type commit
+  (** Type for commit identifiers. Similar to Git's commit SHA1s. *)
+
+  type branch
+  (** Type for persistent branch names. Branches usually share a
+      common global namespace and it's the user's responsibility to
+      avoid name clashes. *)
+
+  type slice
+  (** Type for store slices. *)
+
+  (** Repositories. *)
+  module Repo: sig
+
+    (** {1 Repositories}
+
+        A repository contains a set of branches. *)
+
+    type t
+    (** The type of repository handles. *)
+
+    val v: config -> t Lwt.t
+    (** [v mk_task config] connects to a repository in a
+        backend-specific manner. *)
+
+    val heads: t -> commit list Lwt.t
+    (** [heads] is {!Head.list}. *)
+
+    val branches: t -> branch list Lwt.t
+    (** [branches] is {Branch.list}. *)
+
+    val export: ?full:bool -> ?depth:int ->
+      ?min:commit list -> ?max:commit list ->
+      t -> slice Lwt.t
+    (** [export t ~depth ~min ~max] exports the store slice between
+        [min] and [max], using at most [depth] history depth (starting
+        from the max).
+
+        If [max] is not specified, use the current [heads]. If [min] is
+        not specified, use an unbound past (but can still be limited by
+        [depth]).
+
+        [depth] is used to limit the depth of the commit history. [None]
+        here means no limitation.
+
+        If [full] is set (default is true), the full graph, including the
+        commits, nodes and contents, is exported, otherwise it is the
+        commit history graph only. *)
+
+    val import: t -> slice -> [`Ok | `Error] Lwt.t
+    (** [import t s] imports the contents of the slice [s] in [t]. Does
+        not modify branches. *)
+
+    val task_of_commit: t -> commit -> task option Lwt.t
+    (** [task_of_commit t c] is the description of the commit
+        [c]. Useful to retrieve the commit date and the committer
+        name. Return [None] if [c] is not present in [t]. *)
+
+  end
+
+  val empty: Repo.t -> t Lwt.t
+  (** [empty repo task] is a temporary, empty store. Becomes a
+      normal temporary store after the first update. *)
+
+  val master: Repo.t -> t Lwt.t
+  (** [master repo] is a persistent store based on [r]'s master
+      branch. This operation is cheap, can be repeated multiple
+      times. *)
+
+  val of_branch: Repo.t -> branch -> t Lwt.t
+  (** [of_branch r name] is a persistent store based on the branch
+      [name]. Similar to [master], but use [name] instead
+      {!Branch.S.master}. *)
+
+  val of_commit: Repo.t -> commit -> t Lwt.t
```

Comment:
Is this a commit or a commit ID? If it's a commit ID, I guess it should be returning an `option` (and maybe change the type name to be less confusing). If it's a commit, I'm not sure why we need to provide the repo (since a commit presumably knows where it's from).